lib/services/types/java.service.js

```diff
@@ -1,66 +1,100 @@
-    return javaClass
+ * Process model attributes and generates a template properties
+ * @param {*} model the sequelize model
+ * @returns the template properties of the given model
+ */
+function processAttributes(model) {
+    const properties = [];
+    const imports = [];
+    forEach(model.rawAttributes, (value, key) => {
+        const processed = processType(value.type.constructor.name);
+        if (!processed) return;
+        properties.push(
+            {
+                type: processed.type,
+                name: camelCase(key),
+                defaultValue: processed.defaultValue,
+                methodName: upperFirst(camelCase(key))
+            });
+        if (!!processed.imports.length) {
+            imports.push(...processed.imports);
+        }
+    });
+
+    forEach(model.associations, (item) => {
+        if (item.isMultiAssociation) {
+            properties.push(
+                {
+                    type: `LinkedList<${item.target.name}>`,
+                    name: camelCase(item.as),
+                    defaultValue: "new LinkedList<>()",
+                    methodName: upperFirst(camelCase(item.as))
+                });
+            imports.push("java.util.LinkedList;")
+        } else {
+            properties.push({ type: item.as, name: item.as });
+        }
+    });
+
+    return { properties, imports };
 }
 
 /**
- * Util method to return the equivalent data types from Sequelize types to Java types
- * @param {*} type sequelize field's type 
- * @returns the equivalent Java type
- */
-function getJavaDataType(type) {
+* Util method to return the equivalent data types from Sequelize types to Java types and imports
+* @param {*} type sequelize field's type
+* @returns an object with the Java type and optional imports statements
+*/
+function processType(type) {
+    const result = { type: undefined, imports: [], defaultValue: undefined };
     switch (type) {
-        case 'INTEGER':
-            return 'int'
-        case 'DECIMAL':
-        case 'DOUBLE':
-            return 'double'
-        case 'FLOAT':
-            return 'float'
-        case 'DATE':
-        case 'DATEONLY':
-            return 'java.util.Date'
-        case 'BOOLEAN':
-            return 'boolean'
-        case 'STRING':
-        case 'TEXT':
-            return 'String'
+        case "INTEGER":
+            result.type = "int";
+            result.defaultValue = 0;
+            break;
+        case "DECIMAL":
+        case "DOUBLE":
+            result.type = "double";
+            result.defaultValue = 0.0;
+            break;
+        case "FLOAT":
+            result.type = "float";
+            result.defaultValue = 0.0;
+            break;
+        case "BOOLEAN":
+            result.type = "boolean";
+            result.defaultValue = false;
+            break;
+        case "STRING":
+        case "TEXT":
+            result.type = "String";
+            result.defaultValue = "";
```

Comment:
This should be either '""' or null rather than "" and cause a syntax error;

Reply:
![image](https://github.com/easygrating/sequelize-model-to-file/assets/22077083/1b4c5849-5313-4558-8134-31bdfb82ad2f)